contracts/contracts/strategies/balancer/BalancerMetaPoolStrategy.sol

```diff
@@ -0,0 +1,477 @@
+    )
+        InitializableAbstractStrategy(_stratConfig)
+        BaseBalancerStrategy(_balancerConfig)
+        BaseAuraStrategy(_auraRewardPoolAddress)
+    {}
+
+    /**
+     * @notice Deposits an `_amount` of vault collateral assets
+     * from the this strategy contract to the Balancer pool.
+     * @param _asset Address of the Vault collateral asset
+     * @param _amount The amount of Vault collateral assets to deposit
+     */
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        whenNotInVaultContext
+        onlyVault
+        nonReentrant
+    {
+        address[] memory assets = new address[](1);
+        uint256[] memory amounts = new uint256[](1);
+        assets[0] = _asset;
+        amounts[0] = _amount;
+
+        _deposit(assets, amounts);
+    }
+
+    /**
+     * @notice Deposits specified vault collateral assets
+     * from the this strategy contract to the Balancer pool.
+     * @param _assets Address of the Vault collateral assets
+     * @param _amounts The amount of each asset to deposit
+     */
+    function deposit(address[] memory _assets, uint256[] memory _amounts)
+        external
+        whenNotInVaultContext
+        onlyVault
+        nonReentrant
+    {
+        _deposit(_assets, _amounts);
+    }
+
+    /**
+     * @notice Deposits all supported assets in this strategy contract to the Balancer pool.
+     */
+    function depositAll()
+        external
+        override
+        whenNotInVaultContext
+        onlyVault
+        nonReentrant
+    {
+        uint256 assetsLength = assetsMapped.length;
+        address[] memory assets = new address[](assetsLength);
+        uint256[] memory amounts = new uint256[](assetsLength);
+
+        // For each vault collateral asset
+        for (uint256 i = 0; i < assetsLength; ++i) {
+            assets[i] = assetsMapped[i];
+            // Get the asset balance in this strategy contract
+            amounts[i] = IERC20(assets[i]).balanceOf(address(this));
+        }
+        _deposit(assets, amounts);
+    }
+
+    function _deposit(address[] memory _assets, uint256[] memory _amounts)
+        internal
+    {
+        (IERC20[] memory tokens, , ) = balancerVault.getPoolTokens(
+            balancerPoolId
+        );
+
+        uint256[] memory mappedAmounts = new uint256[](tokens.length);
+        address[] memory mappedAssets = new address[](tokens.length);
+        for (uint256 i = 0; i < _assets.length; ++i) {
+            address asset = _assets[i];
+            uint256 amount = _amounts[i];
+            mappedAssets[i] = toPoolAsset(_assets[i]);
+
+            if (amount > 0) {
+                emit Deposit(asset, platformAddress, amount);
+
+                // wrap rebasing assets like stETH and frxETH to wstETH and sfrxETH
+                (, mappedAmounts[i]) = wrapPoolAsset(asset, amount);
+            }
+        }
+
+        // TODO move this loop into the previous loop
```

Comment:
TODO comment here. Do we need to do it?

Reply:
that was me. No, it's not needed. It's just a gas optimisation which probably wouldn't save that much relative to the total gas cost of a deposit

Reply:
no, its a gas optimisation that is not needed

Reply:
Let's remove the comment then.

Reply:
🗑️